.github/workflows/ci.yml

```diff
@@ -1,20 +1,73 @@
 name: CI
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - "**"
+      - "!master"
```

Comment:
This stuff and above is to
1. Run CI on PRs and new commits
2. Run CI on master branch after any merge into master.

Reply:
Got it. Would this achieve the same effect? When a PR is merged to master, it will trigger a push event (see [this](https://github.community/t/trigger-workflow-only-on-pull-request-merge/17359/3)).

```
  pull_request:
    branches:
      - master # It might be even better to just use "*" , because we usually do PR against a feature branch
    types: [opened, synchronize]

  push:
    branches:
      - "**"
```

Reply:
Both solutions were a bit incorrect causing 2 CI processes at the same time. I think I fixed it.

Reply:
This works when a PR is created from a branch in maticnetwork/bor, but it doesn't seem to run any CI test if the PR is created from a fork repo of bor. I added a new dummy commit on top of your branch, pushed this branch to my fork of bor, and created [this PR](https://github.com/maticnetwork/bor/pull/400). The CI in this PR was skipped, which is not a desired behavior.

Reply:
That solution triggers 2 CI processes for each commit.